lib/active_storage/service/open_stack_service.rb

```diff
@@ -136,6 +146,8 @@ def format_range(range)
       " bytes=#{range.begin}-#{range.exclude_end? ? range.end - 1 : range.end}"
     end
 
-
+    def guess_content_type(io)
```

Comment:
While running this method with safe navigation, specifically: io&.orginial_filename, it raised an error. which means the method #original_filename does not exist on the io param passed.
Same with :content_type.
Did you test it? This method is used only for uploading. which obviously made the whole test suite fail, since in every setup, a file is uploaded.
I'll figure it out. Thanks though :)

Reply:
Yes, `original_filename` and `content_type` may be part of `io` (e.g. when it is a Rack::Test::UploadedFile or Rack::Multipart::UploadedFile) but some IO-objects (File, StringIO,...) do not have those methods. `Marcel::MimeType.for` can still find a mime type in these cases (via magic bytes) but it might be off (e.g. discovering `text/plain` for a `application/json` file). 

That is why we have this unstandardized `change_content_type` method that we wire up in `lib/active_storage/openstack/railtie.rb` to be called when ActiveStorage has figured out the mime type of the blob. This will fix the initially wrongly identified mime type.

Reply:
Yes, fair enough i figured it out yesterday after merging. `String.IO` is used for testing. I guess the most common use case is via an upload via `Rack::Test::UploadedFile or Rack::Multipart::UploadedFile`
That's why I kept the `try`. before releasing the gem